lib/finch/conn.ex

```diff
@@ -100,4 +105,12 @@ defmodule Finch.Conn do
         {:error, conn, error}
     end
   end
+
+  defp append_data_chunk(%Response{body: nil} = response, data_chunk) do
+    %{response | body: data_chunk}
+  end
+
+  defp append_data_chunk(%Response{body: body} = response, data_chunk) when is_binary(body) do
```

Comment:
I don't feel too strongly about it, but this guard seems a bit superfluous to me. Maybe we should flip these clauses and remove the match on nil? What do you think?

Reply:
I can do that! My intent was to detect misuse of the function, but re-reading this now I realize that I'd be better off having `when is_binary(data_chunk)` as the guard instead. Does flipping the two clauses, and adding that guard work for you?